src/js/resize-manager.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * @file resize-manager.js
+ */
+import window from 'global/window';
+import { debounce } from './utils/fn.js';
+import * as Events from './utils/events.js';
+import mergeOptions from './utils/merge-options.js';
+import Component from './component.js';
+
+class ResizeManager extends Component {
+  constructor(player, options) {
+    let RESIZE_OBSERVER_AVAILABLE = options.ResizeObserver || window.ResizeObserver;
+
+    if (options.ResizeObserver === null) {
+      RESIZE_OBSERVER_AVAILABLE = false;
+    }
+
+    // Only create an element when ResizeObserver isn't available
+    const options_ = mergeOptions({createEl: !RESIZE_OBSERVER_AVAILABLE}, options);
+
+    super(player, options_);
+
+    this.ResizeObserver = options.ResizeObserver || window.ResizeObserver;
+    this.loadListener_ = null;
+    this.resizeObserver_ = null;
+    this.debouncedHandler_ = debounce(() => {
+      this.resizeHandler();
+    }, 100, false, player);
+
+    if (RESIZE_OBSERVER_AVAILABLE) {
+      this.resizeObserver_ = new this.ResizeObserver(this.debouncedHandler_);
+      this.resizeObserver_.observe(player.el());
+
+    } else {
+      this.loadListener_ = () => {
+        if (this.el_.contentWindow) {
+          Events.on(this.el_.contentWindow, 'resize', this.debouncedHandler_);
+        }
+        this.off('load', this.loadListener_);
+      };
+
+      this.on('load', this.loadListener_);
+    }
+  }
+
+  createEl() {
+    return super.createEl('iframe', {
+      className: 'vjs-resize-manager'
+    });
+  }
```

Comment:
I believe the element will be created even when it's not needed. Should we suppress the creation of it when `ResizeObserver` is available?